### ISSUE TITLE
solver.as_xarray fails with multidimensional states

### ISSUE THREAD

**Issue by astoeriko** [aseyboldt/sunode#13]
Converting the solution of an ODE with multidimensional states to an xarray dataset fails if I choose `unstack_state=True`:

```python
import numpy as np
import matplotlib.pyplot as plt
import sunode


def multi_dim_ode(t, y, p):
    return {"y": -p.a * y.y}


params = {
    "a": (),
}
states = {
    "y": ("ydim",),
}

problem = sunode.SympyProblem(
    params=params,
    states=states,
    rhs_sympy=multi_dim_ode,
    derivative_params=(),
    coords={"ydim": np.array([0, 2])},
)
solver = sunode.solver.Solver(problem, compute_sens=False, solver="BDF")

y0 = np.zeros((), dtype=problem.state_dtype)
y0["y"] = np.array([1, 3])
tvals = np.linspace(0, 10, num=5)
solver.set_params_dict(
    {
        "a": 0.1,
    }
)

output = solver.make_output_buffers(tvals)
solver.solve(t0=0, tvals=tvals, y0=y0, y_out=output)

# Works
solver.as_xarray(tvals, output, unstack_state=False)

# Does not work
solver.as_xarray(tvals, output, unstack_state=True)
```
It seems that the additional dimensions are not passed to xarray. The error I get is:

```python-traceback
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
~/miniconda3/envs/sunode/lib/python3.8/site-packages/xarray/core/variable.py in as_variable(obj, name)
    101         try:
--> 102             obj = Variable(*obj)
    103         except (TypeError, ValueError) as error:

~/miniconda3/envs/sunode/lib/python3.8/site-packages/xarray/core/variable.py in __init__(self, dims, data, attrs, encoding, fastpath)
    303         self._data = as_compatible_data(data, fastpath=fastpath)
--> 304         self._dims = self._parse_dimensions(dims)
    305         self._attrs = None

~/miniconda3/envs/sunode/lib/python3.8/site-packages/xarray/core/variable.py in _parse_dimensions(self, dims)
    495         if len(dims) != self.ndim:
--> 496             raise ValueError(
    497                 "dimensions %s must have the same length as the "

ValueError: dimensions ('time',) must have the same length as the number of data dimensions, ndim=2

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<ipython-input-13-f84c5315b76f> in <module>
     40 
     41 # Does not work
---> 42 solver.as_xarray(tvals, output, unstack_state=True)

~/git_repos/sunode/sunode/solver.py in as_xarray(self, tvals, out, sens_out, unstack_state, unstack_params)
    324 
    325     def as_xarray(self, tvals, out, sens_out=None, unstack_state=True, unstack_params=True):
--> 326         return self._problem.solution_to_xarray(
    327             tvals, out, self._user_data,
    328             sensitivity=sens_out,

~/git_repos/sunode/sunode/problem.py in solution_to_xarray(self, tvals, solution, user_data, sensitivity, unstack_state, unstack_params)
    127             for name in state:
    128                 assert name not in data
--> 129                 data[name] = ('time', state[name])
    130         else:
    131             data['solution'] = ('time', solution)

~/miniconda3/envs/sunode/lib/python3.8/site-packages/xarray/core/dataset.py in __setitem__(self, key, value)
   1272             )
   1273 
-> 1274         self.update({key: value})
   1275 
   1276     def __delitem__(self, key: Hashable) -> None:

~/miniconda3/envs/sunode/lib/python3.8/site-packages/xarray/core/dataset.py in update(self, other, inplace)
   3573         """
   3574         _check_inplace(inplace)
-> 3575         merge_result = dataset_update_method(self, other)
   3576         return self._replace(inplace=True, **merge_result._asdict())
   3577 

~/miniconda3/envs/sunode/lib/python3.8/site-packages/xarray/core/merge.py in dataset_update_method(dataset, other)
    862                     other[key] = value.drop_vars(coord_names)
    863 
--> 864     return merge_core([dataset, other], priority_arg=1, indexes=dataset.indexes)

~/miniconda3/envs/sunode/lib/python3.8/site-packages/xarray/core/merge.py in merge_core(objects, compat, join, priority_arg, explicit_coords, indexes, fill_value)
    550         coerced, join=join, copy=False, indexes=indexes, fill_value=fill_value
    551     )
--> 552     collected = collect_variables_and_indexes(aligned)
    553 
    554     prioritized = _get_priority_vars_and_indexes(aligned, priority_arg, compat=compat)

~/miniconda3/envs/sunode/lib/python3.8/site-packages/xarray/core/merge.py in collect_variables_and_indexes(list_of_mappings)
    275                 append_all(coords, indexes)
    276 
--> 277             variable = as_variable(variable, name=name)
    278             if variable.dims == (name,):
    279                 variable = variable.to_index_variable()

~/miniconda3/envs/sunode/lib/python3.8/site-packages/xarray/core/variable.py in as_variable(obj, name)
    103         except (TypeError, ValueError) as error:
    104             # use .format() instead of % because it handles tuples consistently
--> 105             raise error.__class__(
    106                 "Could not convert tuple of form "
    107                 "(dims, data[, attrs, encoding]): "

ValueError: Could not convert tuple of form (dims, data[, attrs, encoding]): ('time', array([[1.        , 3.        ],
       [0.77880078, 2.33640235],
       [0.60653066, 1.81959198],
       [0.47236655, 1.41709966],
       [0.36787944, 1.10363832]])) to Variable.


```

**Comment by aseyboldt** [comment]
Thanks for reporting.
This is fixed in d3fe38534538659a4c5691e624df376db5c16567 on master

**Comment by astoeriko** [comment]
Thanks for fixing. :-)
